Description: Change encoding to use the system locale instead of ascii.
Author: einynot <111812541+einynot@users.noreply.github.com>
Origin: backport, https://github.com/aws/amazon-ec2-hibinit-agent/commit/2392e860c7a75c2b61baef749bb90e79a2a3e81e
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/ec2-hibinit-agent/+bug/2024505
Last-Update: 2023-06-20
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/agent/hibinit-agent
+++ b/agent/hibinit-agent
@@ -145,7 +145,7 @@
 
 def find_device_for_file(filename):
     # Find the mount point for the swap file ('df -P /swap')
-    df_out = check_output(['df', '-P', filename]).decode('ascii')
+    df_out = check_output(['df', '-P', filename]).decode(sys.getfilesystemencoding())
     dev_str = df_out.split("\n")[1].split()[0]
     return dev_str
 
